### ISSUE TITLE
Calendar: An error appears in the JS console when an invalid date is entered into a p:calendar textbox

### ISSUE THREAD

**Issue by ngriffin7a** [primefaces/primefaces#6138]
**Describe the defect**

An error appears in the JS console when an invalid date is entered into a p:calendar textbox.

This is a regression in PrimeFaces 8.0 -- it does not happen with PrimeFaces 7.0.

**Reproducer**
- Build the attached project and deploy the war to tomcat/webapps

**Environment:**
 - PF Version: 8.0
 - JSF + version: Mojarra 2.2.20
 - Affected browsers: ALL

**To Reproduce**
- Visit http://localhost:8080/calendar-issue-1.0.0/calendar.jsf
- Show the JS console
- Pick a date with the mouse
- Enter `12/34/5678` into the field and tab-out (blur) the field

**Expected behavior**
There will be no JS errors in the console.

**Defect behavior**

If the defect still exists, then the following error will appear in the JS console:
```
VM210:1 Uncaught SyntaxError: Unexpected token '?'
    at b (jquery.js.jsf?ln=primefaces&v=8.0:2)
    at Ie (jquery.js.jsf?ln=primefaces&v=8.0:2)
    at k.fn.init.replaceWith (jquery.js.jsf?ln=primefaces&v=8.0:2)
    at Object.updateElement (core.js.jsf?ln=primefaces&v=8.0:18)
    at Object.doUpdate (core.js.jsf?ln=primefaces&v=8.0:18)
    at Object.handle (core.js.jsf?ln=primefaces&v=8.0:18)
    at Object.<anonymous> (core.js.jsf?ln=primefaces&v=8.0:18)
    at c (jquery.js.jsf?ln=primefaces&v=8.0:2)
    at Object.fireWith [as resolveWith] (jquery.js.jsf?ln=primefaces&v=8.0:2)
    at l (jquery.js.jsf?ln=primefaces&v=8.0:2)
```

**Example XHTML**
```<!DOCTYPE html>
<html xmlns="http://www.w3c.org/1999/xhtml"
	  xmlns:f="http://java.sun.com/jsf/core"
	  xmlns:h="http://xmlns.jcp.org/jsf/html"
	  xmlns:p="http://primefaces.org/ui">
<h:head/>
<h:body>
	<h:form id="f1">
		<h:panelGroup id="dateOfBirthField" layout="block">
			<p:calendar id="dateOfBirth" label="Date of Birth"
						required="true"
						pattern="MM/dd/yyyy" showOn="button"
						value="#{calendarBacking.dateOfBirth}">
				<f:convertDateTime pattern="MM/dd/yyyy"/>
				<p:ajax update="dateOfBirthField"/>
				<p:ajax event="dateSelect" update="dateOfBirthField"/>
			</p:calendar>
		</h:panelGroup>
		<hr/>
		<p:commandButton action="#{calendarBacking.submit}" update="f1"
						 value="Submit"/>
	</h:form>
</h:body>
</html>
```

**Example Bean**
```package org.liferayfaces.calendarissue;

import javax.faces.bean.ManagedBean;
import java.util.Date;

@ManagedBean
public class CalendarBacking {

	private Date dateOfBirth;

	public Date getDateOfBirth() {
		return dateOfBirth;
	}

	public void setDateOfBirth(Date dateOfBirth) {
		this.dateOfBirth = dateOfBirth;
	}

	public void submit() {
		System.out.printf("Submitted");
	}
}
```

**Comment by ngriffin7a** [comment]
[calendar-issue.zip](https://github.com/primefaces/primefaces/files/4904585/calendar-issue.zip)

**Comment by ngriffin7a** [comment]
[calendar-issue.zip](https://github.com/primefaces/primefaces/files/4904615/calendar-issue.zip)

**Comment by melloware** [comment]
Your problem is this is an invalid EL expression
```xml
<f:convertDateTime pattern="#{MM/dd/yyyy}"/>
```

Change to:
```xml
<f:convertDateTime pattern="MM/dd/yyyy"/>
```
... and it works fine

**Comment by ngriffin7a** [comment]
@melloware I corrected my EL expression after my original submission, but before your comment. I also uploaded a corrected reproducer. I kindly ask that you re-open this issue. Thank you.

**Comment by melloware** [comment]
Reproducer:
[pf-6138.zip](https://github.com/primefaces/primefaces/files/4904673/pf-6138.zip)

I can't reproduce run `mvn clean jetty:run-exploded` and navigate to http://localhost:8080/primefaces-test/test.xhtml your example works fine?

**Comment by melloware** [comment]
Ahh nevermind I see it.

**Comment by melloware** [comment]
Its failing the server side check for a valid date....
![image](https://user-images.githubusercontent.com/4399574/87185092-27c13a80-c2b7-11ea-9b4f-fa4be48b16e4.png)

But when it goes to build the JS the date is invalid so the component breaks.

**Comment by melloware** [comment]
OK I verified its already fixed in 9.0. Try this reproducer which is set for GitHub Mastr snapshot.
[pf-6138.zip](https://github.com/primefaces/primefaces/files/4904732/pf-6138.zip)

**Comment by melloware** [comment]
Confirmed its also Fixed in Elite 8.0.2 with this issue: https://github.com/primefaces/primefaces/issues/5923

**Comment by ngriffin7a** [comment]
@melloware Thanks so much for confirming that it's fixed in Elite 8.0.2 and 9.0. Is there a tentative release schedule for 9.0?

**Comment by melloware** [comment]
Well i don't work for PrimeTek but base on the last 3 releases they are about once a year.  8.0 was released in March so I would guess 9.0 sometime early 2021 if their pattern holds...
